test: fix require_fs_type any and RUNTESTS -f

tomaszkapela opened the pull request:
Fix running tests requiring fs type to be any and forcing a specific fs
through RUNTESTS.

Ref: pmem/issues#468

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/pmem/nvml/1712)
<!-- Reviewable:end -->

janekmi commented:
Reviewed 4 of 4 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/RUNTESTS.ps1, line 189 at r1](https://reviewable.io:443/reviews/pmem/nvml/1712#-KeDVHffgnCkXuGcwOqP:-KeDVHfgJutXNxcy6eIR:b-kpl6h5) ([raw file](https://github.com/pmem/nvml/blob/8151f729726c6ccff6b99a7e072e1c6c48107f7c/src/test/RUNTESTS.ps1#L189)):*
> ```PowerShell
>     # FS type was forced by user
>     if($fstype -ne "all") {
>         $Env:FS_SET = 1
> ```

If this variables do the same please name them the same. FORCE_FS or FS_SET.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1712)*
<!-- Sent from Reviewable.io -->

tomaszkapela commented:
Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/test/RUNTESTS.ps1, line 189 at r1](https://reviewable.io:443/reviews/pmem/nvml/1712#-KeDVHffgnCkXuGcwOqP:-KeDWUADsNI0TtQenr4l:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/8151f729726c6ccff6b99a7e072e1c6c48107f7c/src/test/RUNTESTS.ps1#L189)):*
<details><summary><i>Previously, janekmi (Jan Michalski) wrote…</i></summary><blockquote>

If this variables do the same please name them the same. FORCE_FS or FS_SET.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1712)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Another problem with this patch is that if you specify "-f all", the test requiring "any" will be executed 3 times (2x for pmem + 1x for non-pmem).

---

Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/test/RUNTESTS, line 420 at r2](https://reviewable.io:443/reviews/pmem/nvml/1712#-KedPslqJAkLxlEhE9_V:-KedPslqJAkLxlEhE9_W:b4cc0ss) ([raw file](https://github.com/pmem/nvml/blob/0784703f37b39bcaa7ed4b10030fd866bd7ff25c/src/test/RUNTESTS#L420)):*
> ```
> 		in
> 		none|pmem|non-pmem|any|all)
> 			export FORCE_FS=1
> ```

It's not clear for me why we need this var.

---

*[src/test/unittest/unittest.ps1, line 975 at r2](https://reviewable.io:443/reviews/pmem/nvml/1712#-KedPyDUKCF62PZCr7QR:-KedPyDUKCF62PZCr7QS:b-i2sizn) ([raw file](https://github.com/pmem/nvml/blob/0784703f37b39bcaa7ed4b10030fd866bd7ff25c/src/test/unittest/unittest.ps1#L975)):*
> ```PowerShell
>     sv -Name req_fs_type 1 -Scope Global
>     for ($i=0;$i -lt $args.count;$i++) {
>         # if FS type was forced, treat any as either pmem or non-pmem
> ```

and in other cases we treat "any" as what...?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1712#-:-KedbSxZB-Q8bvVf5ihn:b-f1y82s)*
<!-- Sent from Reviewable.io -->

plebioda commented:
Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/test/RUNTESTS, line 420 at r2](https://reviewable.io:443/reviews/pmem/nvml/1712#-KedPslqJAkLxlEhE9_V:-Ken1J9BKIdkpOLD1we-:b-vi7d3j) ([raw file](https://github.com/pmem/nvml/blob/0784703f37b39bcaa7ed4b10030fd866bd7ff25c/src/test/RUNTESTS#L420)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

It's not clear for me why we need this var.
</blockquote></details>

:+1: any coomment about that in commit message or near the variable ? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1712)*
<!-- Sent from Reviewable.io -->

tomaszkapela commented:
Ugh... who would specify -f all? I feel like all and any don't really mix :)

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/test/RUNTESTS, line 420 at r2](https://reviewable.io:443/reviews/pmem/nvml/1712#-KedPslqJAkLxlEhE9_V:-Kf5xy6FDYFyu_dBSlhQ:btu7squ) ([raw file](https://github.com/pmem/nvml/blob/0784703f37b39bcaa7ed4b10030fd866bd7ff25c/src/test/RUNTESTS#L420)):*
<details><summary><i>Previously, plebioda (Pawel Lebioda) wrote…</i></summary><blockquote>

:+1: any coomment about that in commit message or near the variable ? 
</blockquote></details>

If you have any idea how to fix this without redesigning our test scripts, I'm all ears. I can add a comment, no problem.

---

*[src/test/unittest/unittest.ps1, line 975 at r2](https://reviewable.io:443/reviews/pmem/nvml/1712#-KedPyDUKCF62PZCr7QR:-Kf5yAZCtR6jm9pvwYS0:bbuokhv) ([raw file](https://github.com/pmem/nvml/blob/0784703f37b39bcaa7ed4b10030fd866bd7ff25c/src/test/unittest/unittest.ps1#L975)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

and in other cases we treat "any" as what...?
</blockquote></details>

well... It should be treated exactly the same, as it is written in the README. You mean you don't like the comment?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1712#-:-Kf5xrephZ9g933tlUAg:bc70lf0)*
<!-- Sent from Reviewable.io -->

krzycz commented:
Well, "all" is the default, so you don't need to specify this option, but if you do, the behavior will be different than w/o it.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/test/unittest/unittest.ps1, line 975 at r2](https://reviewable.io:443/reviews/pmem/nvml/1712#-KedPyDUKCF62PZCr7QR:-KfC3gnIhUKmVZLRKPMY:b-or5w93) ([raw file](https://github.com/pmem/nvml/blob/0784703f37b39bcaa7ed4b10030fd866bd7ff25c/src/test/unittest/unittest.ps1#L975)):*
<details><summary><i>Previously, tomaszkapela (Tomasz Kapela) wrote…</i></summary><blockquote>

well... It should be treated exactly the same, as it is written in the README. You mean you don't like the comment?
</blockquote></details>

Yeah, I mean the comment - just skip the first part of the sentence.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1712#-:-KfC5lKd0_p5Y4g74s-B:b-nt17z8)*
<!-- Sent from Reviewable.io -->

tomaszkapela commented:
Fixed it.

---

Review status: 0 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/test/RUNTESTS, line 420 at r2](https://reviewable.io:443/reviews/pmem/nvml/1712#-KedPslqJAkLxlEhE9_V:-KfC8J_7OpgBCZVq9zf7:b1ni3al) ([raw file](https://github.com/pmem/nvml/blob/0784703f37b39bcaa7ed4b10030fd866bd7ff25c/src/test/RUNTESTS#L420)):*
<details><summary><i>Previously, tomaszkapela (Tomasz Kapela) wrote…</i></summary><blockquote>

If you have any idea how to fix this without redesigning our test scripts, I'm all ears. I can add a comment, no problem.
</blockquote></details>

I added a comment why we need this variable in the current design.

---

*[src/test/unittest/unittest.ps1, line 975 at r2](https://reviewable.io:443/reviews/pmem/nvml/1712#-KedPyDUKCF62PZCr7QR:-KfC767GGqitkHPDTD62:b-896fix) ([raw file](https://github.com/pmem/nvml/blob/0784703f37b39bcaa7ed4b10030fd866bd7ff25c/src/test/unittest/unittest.ps1#L975)):*
<details><summary><i>Previously, krzycz (Krzysztof Czurylo) wrote…</i></summary><blockquote>

Yeah, I mean the comment - just skip the first part of the sentence.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1712#-:-KfC7AqcGVJS8rjFQLdq:buyydde)*
<!-- Sent from Reviewable.io -->

krzycz commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/nvml/1712#-:-KfCMQKO1fZijfjvaRmS:bnfp4nl)*
<!-- Sent from Reviewable.io -->
